controllers/snapshot/snapshot_adapter.go

```diff
@@ -88,6 +88,26 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 	return results.ContinueProcessing()
 }
 
+// EnsureGlobalComponentImageUpdated is an operation that ensure the ContainerImage in the Global Candidate List
+// being updated when the ApplicationSnapshot passed all the integration tests
+func (a *Adapter) EnsureGlobalComponentImageUpdated() (results.OperationResult, error) {
+	if gitops.HaveHACBSTestsSucceeded(a.snapshot) {
+		patch := client.MergeFrom(a.component.DeepCopy())
+		for _, component := range a.snapshot.Spec.Components {
+			if component.Name == a.component.Name {
+				a.component.Spec.ContainerImage = component.ContainerImage
```

Comment:
I think that we need to update the a.component.Status.[ContainerImage](https://github.com/redhat-appstudio/application-service/blob/main/api/v1alpha1/component_types.go#L122) here instead of the `Spec.ContainerImage`

Reply:
I might be wrong on this, will update after I get more info

Reply:
Disregard these comments, the `Spec.ContainerImage` is the correct field.